### PR TITLE
Allow setting user password

### LINKED ISSUE #365: Setting password for a user
This endpoint is not covered by the provider, as a work-around for now using the http provider :

```
locals {
  password = "some password"
  userid = "xyz"
  request_headers = {
    "Content-Type"  = "application/json"
    Authorization   = "Bearer ${local.authentik-token}"
  }
}

data "http" "gitea_ldapsearch_password" {
  url    = "https://${var.authentik-url}/api/v3/core/users/${local.userid}/set_password/"
  method = "POST"
  request_headers = local.request_headers
  request_body = jsonencode({password=local.password})
  lifecycle {
    postcondition {
      condition     = contains([201, 204], self.status_code)
      error_message = "Status code invalid"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by BeryJu**
closes #365

### CODE REVIEW

File: internal/provider/resource_user.go [hunk 1]
```diff
@@ -151,6 +181,11 @@ func resourceUserUpdate(ctx context.Context, d *schema.ResourceData, m interface
 	}
 
 	d.SetId(strconv.Itoa(int(res.Pk)))
+
+	diags := resourceUserSetPassword(d, c, ctx)
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply internal/provider/resource_user.go:188]: Good point, added

File: internal/provider/resource_user.go [hunk 2]
```diff
@@ -88,6 +94,28 @@ func resourceUserSchemaToModel(d *schema.ResourceData, c *APIClient) (*api.UserR
 	return &m, nil
 }
 
+func resourceUserSetPassword(d *schema.ResourceData, c *APIClient, ctx context.Context) diag.Diagnostics {
+	password, ok := d.Get("password").(string)
+	if !ok || password == "" {
+		return nil
+	}
+	if !d.HasChange("password") {
```
Comment (rissson) [inline internal/provider/resource_user.go:102]: if you're checking that here, you also have to add a `|| d.IsNewResource()` I think

Comment (rissson) [reply internal/provider/resource_user.go:102]: I meant `&& !d.IsNewResource()`. `!d.HasChange("password")` will evaluate to true on resource creation, and thus the password won't be set

**Comment by senare** [comment]
Hi, I would be needing this, are there anything I can do to help this move forward ?

**Review by rissson — APPROVED** [review]

**Pull request merged**